UltraStar Play/Assets/Common/Util/GameObjectUtils.cs

```diff
@@ -21,4 +24,29 @@ public static T FindComponentWithTag<T>(string tag)
 
         return default(T);
     }
+
+
+    // Looks for a GameObject with the given component, optionally including inactive components.
+    // Note that this is a costly method that should not be called frequently.
+    public static T FindObjectOfType<T>(bool includeInactive) where T : MonoBehaviour
+    {
+        // The current implementation of finding the root transforms is costly.
+        List<Transform> rootTransforms = GameObject.FindObjectsOfType<Transform>().Where(it => it.root == it).ToList();
+        if (includeInactive)
+        {
+            foreach (Transform rootTransform in rootTransforms)
+            {
+                T obj = rootTransform.GetComponentInChildren<T>(true);
+                if (obj != null)
+                {
+                    return obj;
+                }
+            }
+            return null;
```

Comment:
Please write a log entry for these cases and include stacktrace. Maybe this would be a good place to throw an exception.